app/helpers/needs_helper.rb

```diff
@@ -1,14 +1,22 @@
 module NeedsHelper
   def populate_needs
+    rejected = []
+    approved = []
     session[:cart].each do |id, qty|
-      need = NeedType.find(id)
-      current_user.needs.create(name: pluralize(qty, need.name),
-                                description: need.description,
-                                cost: need.cost * qty,
-                                image_url: need.image_url,
-                                slug: need.slug,
-                                category: need.category
-                                )
+      need_type = NeedType.find(id)
+      if current_user.needs.exists?(slug: need_type.slug)
```

Comment:
If the need is already in their "store," the name of the need is added to an array that is passed to the flash message showing what was not allowed into the store.